messagebird/http_client.py

```diff
@@ -48,3 +48,35 @@ def request(self, path, method='GET', params=None):
             response.raise_for_status()
 
         return response_text
+
+    def request_binary(self, path, method='GET', params=None):
+        """Builds a request and gets a response."""
+        if params is None: params = {}
+        url = urljoin(self.endpoint, path)
+
+        headers = {
+            'Accept': 'application/json',
+            'Authorization': 'AccessKey ' + self.access_key,
+            'User-Agent': self.user_agent,
+            'Content-Type': 'application/json'
+        }
+
+        if method == 'DELETE':
```

Comment:
Maybe using one of the ways shown here ? https://data-flair.training/blogs/python-switch-case/

Reply:
It's a (partial) copy-paste from the request method above. In hindsight I can also alter that function to support binary downloads as it currently only supports textual content and then add an additional binary=false parameter.

Reply:
I saw now that its a copy paste, better extract that into a new method using one of ways shown in  https://data-flair.training/blogs/python-switch-case/ instead of multiple if else.

And instead of having a parameter `binary=false`, better have an enum passed as a parameter for how we want the output to be. Now its binary and text, maybe we want more types in the future.